pkg/cmd/cli/cmd/idle.go

```diff
@@ -0,0 +1,557 @@
+	dcGetter := deployclient.New(oclient.RESTClient).DeploymentConfigs(namespace)
+	rcGetter := clientset.FromUnversionedClient(kclient).ReplicationControllers(namespace)
+
+	replicas := make(map[unidlingapi.CrossGroupObjectReference]int32, len(byScalable))
+	toScale := make(map[unidlingapi.CrossGroupObjectReference]scaleInfo)
+
+	mapper, typer := f.Object(false)
+
+	scaleAnnotater := utilunidling.NewScaleAnnotater(delegScaleGetter, dcGetter, rcGetter, func(ref unidlingapi.CrossGroupObjectReference, annotations map[string]string) {
+		annotations[unidlingapi.IdledAtAnnotation] = nowTime.UTC().Format(time.RFC3339)
+	})
+
+	// first, collect the scale info
+	for scaleRef, svcName := range byScalable {
+		obj, scale, err := scaleAnnotater.GetObjectWithScale(scaleRef)
+		if err != nil {
+			fmt.Fprintf(out, "Unable to get scale for %s %s/%s, not marking that scalable as idled...\n", scaleRef.Kind, namespace, scaleRef.Name)
+			svcInfo := byService[svcName]
+			delete(svcInfo.scalables, scaleRef)
+			continue
+		}
+		replicas[scaleRef] = scale.Spec.Replicas
+		toScale[scaleRef] = scaleInfo{scale: scale, obj: obj}
+	}
+
+	// annotate the endpoints objects to indicate which scalables need to be unidled on traffic
+	for serviceName, info := range byService {
+		if info.obj.Annotations == nil {
+			info.obj.Annotations = make(map[string]string)
+		}
+		refsWithScale, err := pairScalesWithIdlables(serviceName, info.obj.Annotations, info.scalables, replicas)
+		if err != nil {
+			fmt.Fprintf(out, "Unable to mark service %s/%s as idled: %v", namespace, serviceName, err)
+			continue
+		}
+
+		if !dryRun {
+			if len(info.scalables) == 0 {
+				fmt.Fprintf(out, "No scalables marked as idled for service %s/%s, not marking as idled...\n", namespace, serviceName)
+				continue
+			}
+
+			metadata, err := meta.Accessor(info.obj)
+			if err != nil {
+				fmt.Fprintf(out, "Unable to mark service %s/%s as idled: %v", namespace, serviceName, err)
+				continue
+			}
+			gvks, _, err := typer.ObjectKinds(info.obj)
+			if err != nil {
+				fmt.Fprintf(out, "Unable to mark service %s/%s as idled: %v", namespace, serviceName, err)
+				continue
+			}
+			oldData, err := json.Marshal(info.obj)
+			if err != nil {
+				fmt.Fprintf(out, "Unable to mark service %s/%s as idled: %v", namespace, serviceName, err)
+				continue
+			}
+
+			mapping, err := mapper.RESTMapping(gvks[0].GroupKind(), gvks[0].Version)
+			if err != nil {
+				fmt.Fprintf(out, "Unable to mark service %s/%s as idled: %v", namespace, serviceName, err)
+				continue
+			}
+
+			if err = setIdleAnnotations(serviceName, info.obj.Annotations, refsWithScale, nowTime); err != nil {
+				fmt.Fprintf(out, "Unable to mark service %s/%s as idled: %v", namespace, serviceName, err)
+				continue
+			}
+			if _, err := patchObj(info.obj, metadata, oldData, mapping, f); err != nil {
+				fmt.Fprintf(out, "Unable to mark service %s/%s as idled: %v", namespace, serviceName, err)
```

Comment:
Every error needs to be counted and at the end and if count > 0 at the end you should return exit code 1.